### ISSUE TITLE
svd2rust creating a fn called `async`

### ISSUE THREAD

**Issue by noslaver** [rust-embedded/svd2rust#441]
When using svd2rust on the efr32mg12p's svd, it creates non-compiling rust code.
It seems this is because the svd contains a register with a field name PRSASYNC, which is later converted into two read-write functions named `async`.

The problematic register:
```xml
<register>
    <name>CH0CTRL</name>
    <description>Channel 0 Control Register</description>
    <addressOffset>0x008</addressOffset>
    <size>32</size>
    <access>read-write</access>
    <resetValue>0x00000000</resetValue>
    <resetMask>0x0000F171</resetMask>
    <fields>
    <field>
        <name>CONVMODE</name>
        <description>Conversion Mode</description>
        <bitOffset>0</bitOffset>
        <bitWidth>1</bitWidth>
        <access>read-write</access>
    </field>
    <field>
        <name>TRIGMODE</name>
        <description>Channel 0 Trigger Mode</description>
        <bitOffset>4</bitOffset>
        <bitWidth>3</bitWidth>
        <access>read-write</access>
        <enumeratedValues>
        <enumeratedValue>
            <name>SW</name>
            <description>Channel 0 is triggered by CH0DATA or COMBDATA write</description>
            <value>0x00000000</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRS</name>
            <description>Channel 0 is triggered by PRS input</description>
            <value>0x00000001</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>REFRESH</name>
            <description>Channel 0 is triggered by Refresh timer</description>
            <value>0x00000002</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>SWPRS</name>
            <description>Channel 0 is triggered by CH0DATA/COMBDATA write or PRS input</description>
            <value>0x00000003</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>SWREFRESH</name>
            <description>Channel 0 is triggered by CH0DATA/COMBDATA write or Refresh timer</description>
            <value>0x00000004</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>LESENSE</name>
            <description>Channel 0 is triggered by LESENSE</description>
            <value>0x00000005</value>
        </enumeratedValue>
        </enumeratedValues>
    </field>
    <field>
        <name>PRSASYNC</name>
        <description>Channel 0 PRS Asynchronous Enable</description>
        <bitOffset>8</bitOffset>
        <bitWidth>1</bitWidth>
        <access>read-write</access>
    </field>
    <field>
        <name>PRSSEL</name>
        <description>Channel 0 PRS Trigger Select</description>
        <bitOffset>12</bitOffset>
        <bitWidth>4</bitWidth>
        <access>read-write</access>
        <enumeratedValues>
        <enumeratedValue>
            <name>PRSCH0</name>
            <description>PRS ch 0 triggers a conversion.</description>
            <value>0x00000000</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH1</name>
            <description>PRS ch 1 triggers a conversion.</description>
            <value>0x00000001</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH2</name>
            <description>PRS ch 2 triggers a conversion.</description>
            <value>0x00000002</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH3</name>
            <description>PRS ch 3 triggers a conversion.</description>
            <value>0x00000003</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH4</name>
            <description>PRS ch 4 triggers a conversion.</description>
            <value>0x00000004</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH5</name>
            <description>PRS ch 5 triggers a conversion.</description>
            <value>0x00000005</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH6</name>
            <description>PRS ch 6 triggers a conversion.</description>
            <value>0x00000006</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH7</name>
            <description>PRS ch 7 triggers a conversion.</description>
            <value>0x00000007</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH8</name>
            <description>PRS ch 8 triggers a conversion.</description>
            <value>0x00000008</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH9</name>
            <description>PRS ch 9 triggers a conversion.</description>
            <value>0x00000009</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH10</name>
            <description>PRS ch 10 triggers a conversion.</description>
            <value>0x0000000A</value>
        </enumeratedValue>
        <enumeratedValue>
            <name>PRSCH11</name>
            <description>PRS ch 11 triggers a conversion.</description>
            <value>0x0000000B</value>
        </enumeratedValue>
        </enumeratedValues>
    </field>
    </fields>
</register>
```

**Comment by noslaver** [comment]
Just saw async sanitizing was merged but unreleased.
